sys-fs/eudev/eudev-3.2.12-r1.ebuild

```diff
@@ -0,0 +1,263 @@
+# Copyright 1999-2023 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+KV_MIN=2.6.39
+
+inherit linux-info multilib-minimal toolchain-funcs udev
+
+if [[ ${PV} = 9999* ]]; then
+	EGIT_REPO_URI="https://github.com/eudev-project/eudev.git"
+	inherit autotools git-r3
+else
+	MY_PV=${PV/_pre/-pre}
+	SRC_URI="https://github.com/eudev-project/eudev/releases/download/v${MY_PV}/${PN}-${MY_PV}.tar.gz"
+	S="${WORKDIR}"/${PN}-${MY_PV}
+
+	if [[ ${PV} != *_pre* ]] ; then
+		KEYWORDS="~alpha ~amd64 ~arm ~arm64 ~hppa ~ia64 ~loong ~m68k ~mips ~ppc ~ppc64 ~riscv ~s390 ~sparc ~x86"
+	fi
+fi
+
+DESCRIPTION="Linux dynamic and persistent device naming support (aka userspace devfs)"
+HOMEPAGE="https://github.com/eudev-project/eudev"
+
+LICENSE="LGPL-2.1 MIT GPL-2"
+SLOT="0"
+IUSE="+kmod rule-generator selinux split-usr static-libs test"
+RESTRICT="!test? ( test )"
+
+DEPEND="
+	>=sys-apps/util-linux-2.20
+	>=sys-kernel/linux-headers-${KV_MIN}
+	virtual/libcrypt:=
+	kmod? ( >=sys-apps/kmod-16 )
+	selinux? ( >=sys-libs/libselinux-2.1.9 )
+	!sys-apps/gentoo-systemd-integration
+	!sys-apps/systemd
+	!>=virtual/libudev-251
```

Comment:
This isn't right. Soft blockers are for file-collisions and hard blockers (to be used even more sparingly) are for when things can't even be temporarily installed at the same time.

Portage does not backtrack around blockers and aren't a valid reverse dependency relation. It leads to confusing output and is not the right tool for the job here.

Reply:
What would be the right solution here? Fix portage?

Because this is entirely caused by an esoteric feature in `=dev-libs/libgudev-238`, which no other package requires, and yet people get told the right solution is to uninstall `sys-fs/eudev`. Seems rather confusing to me?

Reply:
You would have to specify the right thing for this in PMS. There's no construct for this.

But note that since you opened this PR originally, we have:
```
commit 6404b064d63d182da4a8a193533a188cdf832d41
Author: Mike Gilbert <floppym@gentoo.org>
Date:   Sun Jul 30 14:07:47 2023 -0400

    virtual/libudev: add eudev and sticky-tags USE flags

    eudev lacks API support for the new libudev functions that differentiate
    between sticky and current tags on device events.

    Add a USE flag so we can depend on the new API from libgudev.

    Signed-off-by: Mike Gilbert <floppym@gentoo.org>
```

which I think more or less handles it.

Reply:
Seems good. Thank you.